syntax = "proto3";

package roadwork.system;

message SpaceWrapper {
    oneof type {
        SpaceBox box = 1;
        SpaceDiscrete discrete = 2;
        SpaceTuple tuple = 3;
    }
}

message SpaceDiscrete {
    int32 observation = 1;
}

message SpaceBox {
    repeated double observation = 1;
}

message SpaceTuple {
    repeated SpaceWrapper tuples = 1;
}